tests/test_injection.py

```diff
@@ -37,7 +38,13 @@ def __init__(self, crawler):
             self.crawler = crawler
 
         def __call__(self, to_provide):
-            return [cls(content) if content else cls() for cls in classes]
+            result = []
+            for cls in to_provide:
+                obj = cls(content) if content else cls()
+                if metadata := getattr(cls, "__metadata__", None):
+                    obj.__metadata__ = metadata
```

Comment:
We need the same change in all real providers if we want to support annotating their provided classes.

Reply:
Another problem with this is when `obj` has `__slots__` and this breaks.

Reply:
As we only return instances from the provider's `__call__`, and we need to specify which of them had which annotations, an alternative to this code would be changing the `__call__` interface to e.g. return a tuple.